### ISSUE TITLE
3rd person view results in graphical glitches or freeze

### ISSUE THREAD

**Issue by Cat-McCatface** [TheTemportalist/EsoTeriCraft#9]
Since I'm waiting for content to update to 1.9.4 I decided to do a play around by morphing into a cow and running around with camera in 3rd person. Game promptly decided nope and started visual glitches like other animals getting dinnerbone/upsidedown effect and other models merging with player model. Another issue was I was unable to see the pause menu while in 3rd person. The error in log was the following which I thought might be effect of Better Forestry mod but it appears to use vanilla cow model.
###### #### GL ERROR

[20:48:49] [Client thread/ERROR]: @ Post render
[20:48:49] [Client thread/ERROR]: 1283: Stack overflow
[20:48:49] [Client thread/ERROR]: Couldn't render entity
java.lang.ClassCastException: net.minecraft.client.model.ModelCow cannot be cast to net.minecraft.client.model.ModelBiped
    at net.minecraft.client.renderer.entity.layers.LayerHeldItem.func_188358_a(LayerHeldItem.java:59) ~[bts.class:?]
    at net.minecraft.client.renderer.entity.layers.LayerHeldItem.func_177141_a(LayerHeldItem.java:42) ~[bts.class:?]
    at net.minecraft.client.renderer.entity.RenderLivingBase.func_177093_a(RenderLivingBase.java:486) ~[bsa.class:?]
    at net.minecraft.client.renderer.entity.RenderLivingBase.func_76986_a(RenderLivingBase.java:209) [bsa.class:?]
    at net.minecraft.client.renderer.entity.RenderLiving.func_76986_a(RenderLiving.java:50) [bsd.class:?]
    at net.minecraft.client.renderer.entity.RenderLiving.func_76986_a(RenderLiving.java:16) [bsd.class:?]
    at temportalist.esotericraft.emulation.client.EntityModel.forceRender(EntityModel.scala:27) [EntityModel.class:?]
    at temportalist.esotericraft.transmorigification.client.Client$.renderPlayerPre(Client.scala:197) [Client$.class:?]
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_125_Client$_renderPlayerPre_Pre.invoke(.dynamic) [?:?]
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:49) [ASMEventHandler.class:?]
    at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:?]
    at net.minecraft.client.renderer.entity.RenderPlayer.func_76986_a(RenderPlayer.java:58) [buh.class:?]
    at net.minecraft.client.renderer.entity.RenderPlayer.func_76986_a(RenderPlayer.java:28) [buh.class:?]
    at net.minecraft.client.renderer.entity.RenderManager.func_188391_a(RenderManager.java:378) [brj.class:?]
    at net.minecraft.client.renderer.entity.RenderManager.func_188388_a(RenderManager.java:362) [brj.class:?]
    at net.minecraft.client.renderer.RenderGlobal.func_180446_a(RenderGlobal.java:800) [bnl.class:?]
    at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1726) [bnd.class:?]
    at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1555) [bnd.class:?]
    at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1334) [bnd.class:?]
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1077) [bcd.class:?]
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:369) [bcd.class:?]
    at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
    at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
[20:48:49] [Client thread/ERROR]: 

after this I fired it up again to try to reproduce it. morphed into creeper and didn't have any issues. killed a cow again and attempted a morph which resulted in a lockup of the client. vanilla launcher didn't stop the event to produce a proper log.
log: http://pastebin.com/M782Yym0

I made a fresh world and spawned a vanilla cow with an egg just to be sure but wasn't able to reproduce the issue with that animal type.

EsoTeriCraft-1.9.4-0.0.1.73
Origin-1.9.4-9.1.8
Better Agriculture 1.9-0.13 (mod containing cow entity in question)
Forge: 1954 (1.9.4)

**Comment by TheTemportalist** [comment]
Ha. Haha. Hahahaha. That's..... Fun. I'll look into this soon...

**Comment by TheTemportalist** [comment]
@Cat-McCatface can you reproduce without Better Agriculture?

**Comment by Cat-McCatface** [comment]
So far I can't reproduce it with non vanilla mobs although. Seems limited to just entities in Better AG.

wither cat (ender zoo) has odd animation (front legs using arm like animation), rabbit has no walking animation at all, horse has double speed animation, squid only animates once, guardian animation is also hyper, enderman is pretty slow walking speed, spider walking animation fast.

No problems with entities from Lycanites Mobs although that mod has some issues of its own [mob variant related].
(sorry this wasn't helpful, I just went through most of the vanilla mobs with spawn eggs and spammed f5 to see what happens when I fly around)

**Comment by Cat-McCatface** [comment]
I'm going to file a bug report with Better AG in case something that dev can do about it.

**Comment by knoxhack** [comment]
Yeah, thats my mod causing that, I will have to take a look at EsoTeriCraft's code so I can add support for it. Most of my animal entities are still WIP.

**Comment by Cat-McCatface** [comment]
Thanks

**Comment by knoxhack** [comment]
@TheTemportalist Any info or tip on how to fix this and how I can help to make my mobs compatible with your mod would be helpful :)

**Comment by TheTemportalist** [comment]
Hey, thanks for being thorough friends! As far as compatibility, just try not to be hacky. I'm aware there are some render bugs. Soon I would like to implement an API for rendering, with which mods can add specific custom rendering code. Until then, I'm not sure what the issue could be considering I pass the entity cores pounding wth each render to the renderer.
